### PR TITLE
Dtd 383

### PR DESCRIPTION
**Pull request by modivilioglu**

### CODE REVIEW

**Review by jayasreemca — APPROVED** [review]

File: app/uk/gov/hmrc/timetopayproxy/models/BreathingSpace.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+package uk.gov.hmrc.timetopayproxy.models
+import java.time.LocalDate
+
+import play.api.libs.json.Json
+
+case class BreathingSpace(debtRespiteFrom: LocalDate,
+                          debtRespiteTo: LocalDate,
+                          paymentDate: LocalDate,
+                          paymentAmount: BigDecimal)
+
```
Comment (roccobruno) [inline app/uk/gov/hmrc/timetopayproxy/models/BreathingSpace.scala:26]: don't understand this model.. Why does BS have a payment data?

**Review by roccobruno — COMMENTED** [review]
2 small comments

File: app/uk/gov/hmrc/timetopayproxy/models/BreathingSpace.scala (hunk 1 above)
Comment (modivilioglu) [reply app/uk/gov/hmrc/timetopayproxy/models/BreathingSpace.scala:26]: Hi, Regarding the spec v.5 the payload was as follows:
```"breathingSpaces": {
                      "type": "array",
                      "items": [
                        {
                          "type": "object",
                          "properties": {
                            "debtRespiteFrom": {
                              "type": "string",
                              "description": "Breathing space from date"
                            },
                            "debtRespiteTo": {
                              "type": "string",
                              "description": "Breathing space to date"
                            },
                            "paymentDate": {
                              "type": "string",
                              "descprition": ""
                            },
                            "paymentAmount": {
                              "type": "integer",
                              "descprition": ""
                            }
                          }
                        }
                      ]
```

File: app/uk/gov/hmrc/timetopayproxy/connectors/TtpConnector.scala [hunk 2]
```diff
@@ -19,19 +19,25 @@ package uk.gov.hmrc.timetopayproxy.connectors
...
-
 import javax.inject.{Inject, Singleton}
 import uk.gov.hmrc.timetopayproxy.config.AppConfig
 
 @ImplementedBy(classOf[DefaultTtpConnector])
 trait TtpConnector {
   def generateQuote(ttppRequest: GenerateQuoteRequest)(implicit ec: ExecutionContext, hc: HeaderCarrier): TtppEnvelope[GenerateQuoteResponse]
 
-  def getExistingQuote(customerReference: String, pegaId: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): TtppEnvelope[RetrievePlanResponse]
+  def getExistingQuote(customerReference: CustomerReference, pegaId: PegaPlanId)(implicit ec: ExecutionContext, hc: HeaderCarrier): TtppEnvelope[RetrievePlanResponse]
```
[the remark this answers is not in the document]
Comment (modivilioglu) [reply app/uk/gov/hmrc/timetopayproxy/connectors/TtpConnector.scala:33]: Done. Sent a new PR as this one had already been merged.
https://github.com/hmrc/time-to-pay-proxy/pull/10

**Pull request merged**